rust/kernel/src/file_operations.rs

```diff
@@ -41,16 +41,34 @@ pub enum SeekFrom {
     Current(i64),
 }
 
+fn unwrap_error<T>(r: KernelResult<T>) -> T
+where
+    T: TryFrom<c_types::c_int>,
+    T::Error: core::fmt::Debug,
+{
+    match r {
+        Ok(v) => v,
+        Err(e) => T::try_from(e.to_kernel_errno()).unwrap(),
+    }
+}
+
+macro_rules! allow_result {
```

Comment:
I like how this looks. How about naming it `handle_kerror` or something?

Reply:
That name also works, although it looks to me like it is converting to a `kerror` instead, by handling a `KernelResult`. Hmm... `from_KernelResult!`? `to_kerror!`? See other suggestions in the other thread too.

Reply:
Thanks for the suggestions! I also liked the idea of using the same name for the function and macro, so I switched to from_KernelResult, but the compiler issues a warning:

```
44 | fn from_KernelResult<T>(r: KernelResult<T>) -> T
   |    ^^^^^^^^^^^^^^^^^ help: convert the identifier to snake case: `from_kernel_result`
```

So how do you all feel about `from_kernel_result`?

Reply:
Ok, I updated it to use `from_kernel_result`. Thanks!

Reply:
Both are fine with me! Since `from_kernel_result` is probably more natural for Rust, let's go with that.